enterprise/tailnet/pgcoord.go

```diff
@@ -123,6 +123,11 @@ func NewPGCoord(ctx context.Context, logger slog.Logger, ps pubsub.Pubsub, store
 	return c, nil
 }
 
+func (c *pgCoord) ServeMultiAgent(id uuid.UUID) agpl.MultiAgentConn {
+	_, _ = c, id
+	panic("not implemented") // TODO: Implement
```

Comment:
this effectively torpedos the `pgCoord`, so we can't merge this PR until fixed.

Reply:
Since I'm still mostly flushing out the API I didn't think it made sense to impl on `pgCoord` yet. Definitely a blocker for merge though.

Reply:
FWIW, I think what we need to do is to drop the `agent_id` column from `tailnet_clients` table, and add a new table that tracks the agents that each client wants to connect to

```sql
CREATE TABLE tailnet_subscriptions (
    client_id uuid NOT NULL,
    coordinator_id uuid NOT NULL,
    agent_id uuid NOT NULL,
    PRIMARY KEY (client_id, coordinator_id, agent_id),
    FOREIGN KEY (client_id) REFERENCES tailnet_clients(id) ON DELETE CASCADE,
    FOREIGN KEY (coordinator_id) REFERENCES tailnet_coordinators(id) ON DELETE CASCADE
);
```